src/sql/workbench/parts/objectExplorer/browser/serverTreeRenderer.ts

```diff
@@ -152,19 +158,72 @@ export class ServerTreeRenderer implements IRenderer {
 		let iconLowerCaseName = iconName.toLocaleLowerCase();
 		templateData.icon.classList.add(iconLowerCaseName);
 
+		if (treeNode.iconPath) {
+			iconRenderer.putIcon(templateData.icon, treeNode.iconPath);
+		}
+
 		templateData.label.textContent = treeNode.label;
 		templateData.root.title = treeNode.label;
 	}
 
+	private getContextKeyService(connection: ConnectionProfile): IContextKeyService {
+		if (!connection || !connection.id) { return undefined; }
+		let serverInfo: ServerInfo = this._connectionManagementService.getServerInfo(connection.id);
+		if (!serverInfo) { return undefined; }
+
+		let scopedContextService: IContextKeyService = this._contextKeyService.createScoped();
+		let serverInfoContextKey = new ServerInfoContextKey(scopedContextService);
```

Comment:
@YurongHe - how did we handle setting isCloud and isBigDataCluster context key info from right-click actions in Object Explorer? Can Gene do the same here to ensure these are set from the extension, instead of setting internally?

Reply:
`mssql:iscloud` and `mssql:iscluster` context keys are loaded in mssql extension after dashboard opens. (extensions\mssql\src\contextProvider.ts)